### DISCUSSION TITLE
splitting Vec<u8> of non unicode characters

### DISCUSSION THREAD

**Discussion by bobi6666** [rust-lang/regex#928] [Q&A] [answered]
hello, I have a u8 vector, there can be non-unicode characters in that vector and I needed to divide it wherever there is a \0, but with the fact that if there is a \n or \r\n, it will not stop but will continue. Can you please show me example?

**Comment by BurntSushi** [comment, marked as answer]
I think you probably want to use [`bytes::Regex`](https://docs.rs/regex/latest/regex/bytes/struct.Regex.html) for this.

As for a specific example, I don't really understand what you're saying. Could you please provide some input and the desired output?

**Comment by bobi6666** [comment]
i would just want example on how to build Regex::new from regex::bytes
crate to cover my question

2022-11-17 13:32 GMT+01:00, Andrew Gallant ***@***.***>:

**Comment by BurntSushi** [comment]
It's right there in the link I gave you. :-) For example: https://docs.rs/regex/latest/regex/bytes/struct.Regex.html#method.find

**Comment by bobi6666** [comment]
yesterday i wanted to build with unicode disabled and when i did
let re = regex::bytes::Regex::new(r"0x0").unwrap();
it didn't work

2022-11-17 14:18 GMT+01:00, Andrew Gallant ***@***.***>:

**Reply by BurntSushi** [reply]
If you can't answer my original question

> Could you please provide some input and the desired output?

then I can't help you. I won't be responding further if your next comment doesn't answer that question. I apologize if I come across as rude, but this has been a common pattern with you in past discussions. They just go back-and-forth endlessly without my understanding what you actually want to do.

All I'm asking for is a simple example that shows the input and what you want your output to be. If that isn't something you know how to provide, then I'm sorry, but I just don't have the time to give the help required here.

**Reply by BurntSushi** [reply]
This might help: https://jvns.ca/blog/good-questions/

(I linked to [ESR's version of the same thing](http://www.catb.org/%7Eesr/faqs/smart-questions.html) previously, but I forgot just how patronizing it was.)

**Comment by bobi6666** [comment]
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=65826762b40d386ae495a1781305930f

2022-11-17 14:45 GMT+01:00, Andrew Gallant ***@***.***>:

**Reply by BurntSushi** [reply]
Thanks, but you still didn't answer my question! You didn't tell me what output you wanted. And the program you've given me has _zero_ output.

Anyway, I took a guess: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d315997c5951b2aca3fd1e974fccbc2a

**Reply by BurntSushi** [reply]
Notice that you don't even need a `bytes::Regex` because a NUL byte is valid UTF-8.

**Comment by bobi6666** [comment]
output will be later Vec<&[u8]>
but about that i am fine and if your regex idea will not stop after
non utf16 characters then you did what i wanted

2022-11-17 15:10 GMT+01:00, Andrew Gallant ***@***.***>:

**Comment by bobi6666** [comment]
I'll try to translate it so you can better understand what I need:
in the example I sent you, there may be bytes that are not utf8 but
may be utf16. I need to split them, but in such a way that they can be
collected at the end into Vec<&[u8]> or Vec<Vec<u8>>, but I believe
that if they are split so their appearance will be preserved as they
are and I will not be blamed if they do not carry utf 8

2022-11-17 15:14 GMT+01:00, Peter Kubek ***@***.***>:

**Reply by BurntSushi** [reply]
A `&str` can be converted to `&[u8]` via [`str::as_bytes`](https://doc.rust-lang.org/std/primitive.str.html#method.as_bytes).

**Comment by bobi6666** [comment]
do you think this split will be faster than normal split from std if
my file that i work with will be big?

2022-11-17 15:39 GMT+01:00, Andrew Gallant ***@***.***>:

**Reply by BurntSushi** [reply]
Maybe. Benchmark it.

**Comment by bobi6666** [comment]
hello, I want to ask you if aho_corasick supports processing non utf8
bytes if i am using replace_all_with method?

2022-11-17 16:08 GMT+01:00, Andrew Gallant ***@***.***>:

**Reply by BurntSushi** [reply]
Yes... Of course...
